src/hip_hcc.cpp

```diff
@@ -1101,9 +1121,18 @@ void ihipCtx_t::locked_waitAllStreams() {
     LockedAccessor_CtxCrit_t crit(_criticalData);
 
     tprintf(DB_SYNC, "waitAllStream\n");
+    bool need_printf_flush = false;
     for (auto streamI = crit->const_streams().begin(); streamI != crit->const_streams().end();
          streamI++) {
-        (*streamI)->locked_wait();
+        bool waited;
+        (*streamI)->locked_wait(waited);
+        need_printf_flush = !waited;
```

Comment:
With a `bool locked_waited()` function, the for loop could be rewritten as:
`for (auto streamI = crit->const_streams().begin(); `
`      streamI != crit->const_streams().end();`
`      ++streamI) { `
`            need_printf_flush |= (*streamI)->locked_waited()`
` }`

Reply:
Using `|=` isn't correct here.  We are only concerned with whether the last stream actually performed the wait+flush.  Using `|=` would result in more print flushes than just checking the last stream.

Reply:
@jeffdaily Thanks for pointing that out. Instead of `|=` the original code's `!` of `(*streamI)->locked_waited()` should work then, indicating whether the last (in the iteration order) non-empty stream had a wait marker, right?
Does the iteration order from `begin()` to `end()` correspond to oldest to newest time order? IOW, would iterating through the streams in reverse order matter?

Reply:
We are only interested in whether the last iteration did a flush through `locked_waited`.  `|=` doesn't provide that information.  The traversal order through the list of stream doesn't matter here.